src/components/GenreSelect/GenreSelect.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+import './GenreSelect.scss'
+
+interface Props {
+  genres: string[];
+  initialSelectedGenre: string;
+  onSelect: (text: string) => void;
+}
+
+function GenreSelect({genres, initialSelectedGenre, onSelect}: Props) {
+  const [selectedGenre, setSelectedGenre] = useState(initialSelectedGenre);
+
+  const handleGenreSelect = (genre: string) => {
+    setSelectedGenre(genre);
+    onSelect(genre);
+  }
+
+  return (
+    <div className="genre-select m-3">
+      {genres.map((genre) => (
+          <button
+              key={genre}
+              onClick={() => handleGenreSelect(genre)}
```

Comment:
It is better to have the function written useCallback in case of `.map`, otherwise the component will be re-rendered each time genre is selected:

```ts
const handleGenreSelect = useCallback((genre: string) => {
  return (event: React.MouseEvent) => {
    setSelectedGenre(genre);
    onSelect(genre);
  }
}, [])

// ...

<button onClick={handleGenreSelect (genre)}>{genre}</button>

```